zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -78,6 +83,58 @@ use input_selection::{
     GreedyInputSelector, GreedyInputSelectorError, InputSelector, InputSelectorError,
 };
 
+/// Represents the identifier for an account that was derived from a ZIP-32 HD seed and account index.
+#[derive(Debug, Clone)]
+pub struct HDSeedAccount(
+    pub HDSeedFingerprint,
+    pub zip32::AccountId,
+    pub UnifiedFullViewingKey,
+);
```

Comment:
All we need for an identifier is `(HDSeedFingerprint, zip32::AccountId)`. Given that this also includes a UFVK, and that in `zcash_client_sqlite` this is not used as an identifier (a separate `u32` newtype is used for that purpose), this should be re-documented (in the context of how `Account` is changed), or refactored to separate out the ID-like part from the `(ID, UFVK)` combination.

Separately, we generally try to avoid public fields in new structs, instead opting for explicit constructors and getters (so that we can change their internals later if necessary without breaking public API changes).

Reply:
Good points. I've redocumented to stop claiming that this is somehow the ID for an account. It is not and wasn't intended to be.

As for avoiding fields, I find it more cumbersome to avoid public fields, and several times in this or other zcash-related crates, the fields were hidden without a function that exposed them, cramping what I needed to accomplish. So I'd personally lean on leaving it as-is. But if it's a merge-blocker, I'll fold.